src/catalog/catalog.cpp

```diff
@@ -921,6 +933,28 @@ std::shared_ptr<TableCatalogObject> Catalog::GetTableObject(
   return table_object;
 }
 
+/**
+ * Drop all the temporary tables associated with the namespace.
+ */
+void Catalog::DropTempTables(const std::string &database_name,
+                             const std::string &session_namespace,
+                             concurrency::TransactionContext *txn) {
+  auto database_object =
+      DatabaseCatalog::GetInstance()->GetDatabaseObject(database_name, txn);
+  //get pg_table
+  auto pg_table =
+      catalog_map_[database_object->GetDatabaseOid()]->GetTableCatalog();
+  // get all the tables to be dropped
+  auto tables_dropped = pg_table->GetTableObjects(session_namespace, txn);
+  // drop all tables.
+  for (auto iter = tables_dropped.begin(); iter != tables_dropped.end();
+       iter++) {
+    // is this a safeway to use?
```

Comment:
What do you mean for "safeway" here?   I suppose it is just the ordinary iterator.